File: start-web.js
#!/usr/bin/env node
import { spawn } from 'child_process';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));

// 安装依赖
console.log('Installing dependencies...');
const installProcess = spawn('pnpm', ['install'], {
  cwd: __dirname,
  stdio: 'inherit'
});

installProcess.on('close', (code) => {
  if (code !== 0) {
    console.error('Installation failed');
    process.exit(1);
  }
  
  // 启动 Expo Web
  console.log('Starting Expo Web...');
  const webProcess = spawn('pnpm', ['run', 'web'], {
    cwd: __dirname,
    stdio: 'inherit'
  });

  webProcess.on('close', (code) => {
    process.exit(code);
  });
});

File: start-web.py
#!/usr/bin/env python3

import subprocess
import os
import sys

project_dir = os.path.dirname(os.path.abspath(__file__))

print("📦 Installing dependencies...")
try:
    subprocess.run(['npm', 'install'], cwd=project_dir, check=False)
    print("✅ Dependencies installed\n")
except Exception as e:
    print(f"⚠️  Installation warning: {e}\n")

print("🚀 Starting Expo Web server...")
try:
    subprocess.run(['npx', 'expo', 'start', '--web'], cwd=project_dir)
except KeyboardInterrupt:
    print("\n✅ Expo Web stopped")
    sys.exit(0)
except Exception as e:
    print(f"❌ Error: {e}")
    sys.exit(1)
